_LowLevelCode/cpp/cpp_communicator/serialiser/c_deserialise.cpp

```diff
@@ -0,0 +1,363 @@
+/*=========================================================
+ * c_deserialise.cpp
+ * Deserialise serialised data back into a MATLAB object
+ *
+ * See also:
+ * hlp_serialize
+ * hlp_deserialize
+ *
+ * This is a MEX-file for MATLAB.
+ *=======================================================*/
+
+#include <iostream>
+#include <cstring>
+#include <cmath>
+#include <vector>
+#include "cpp_serialise.hpp"
+
+size_t memPtr;
```

Comment:
In MPI parallel each process will have their own independent copies. I don't see this really being OMP parallel, but if it were, it would have to be via a manual breakdown of the copy loops and then it would be a case of ensuring it was atomically updated through the enclosing parallel.

Reply:
As long as each process is addressing its own memory and that multiple processes are not accessing the pointer simultaneously, guessing it's fine. Just checking to make sure